src/main/scala/subsystem/Cluster.scala

```diff
@@ -0,0 +1,230 @@
+      domain.crossIntIn(crossingParams.crossingType, node) := context.msipNodes(hartid)
+    }
+
+    domain.element.meipNodes.foreach { case (hartid, node) =>
+      domain.crossIntIn(crossingParams.crossingType, node) := context.meipNodes(hartid)
+    }
+
+    domain.element.seipNodes.foreach { case (hartid, node) =>
+      domain.crossIntIn(crossingParams.crossingType, node) := context.seipNodes(hartid)
+    }
+
+    domain.element.tileToPlicNodes.foreach { case (hartid, node) =>
+      FlipRendering { implicit p =>
+        context.tileToPlicNodes(hartid) :=* domain.crossIntOut(crossingParams.crossingType, node) }
+    }
+    context.ibus.fromSync :=* domain.crossIntOut(crossingParams.crossingType, domain.element.ibus.toPLIC)
+
+    domain.element.nmiNodes.foreach { case (hartid, node) =>
+      node := context.nmiNodes(hartid)
+    }
+  }
+
+  def connectPRC(domain: ClusterPRCIDomain, context: ClusterContextType): Unit = {
+    implicit val p = context.p
+    domain.element.allClockGroupsNode :*= context.allClockGroupsNode
+    domain {
+      domain.element_reset_domain.clockNode := crossingParams.resetCrossingType.injectClockNode := domain.clockNode
+    }
+  }
+
+  def connectOutputNotifications(domain: ClusterPRCIDomain, context: ClusterContextType): Unit = {
+    implicit val p = context.p
+    context.tileHaltXbarNode  :=* domain.crossIntOut(NoCrossing, domain.element.tileHaltXbarNode)
+    context.tileWFIXbarNode   :=* domain.crossIntOut(NoCrossing, domain.element.tileWFIXbarNode)
+    context.tileCeaseXbarNode :=* domain.crossIntOut(NoCrossing, domain.element.tileCeaseXbarNode)
+
+  }
+
+  def connectInputConstants(domain: ClusterPRCIDomain, context: ClusterContextType): Unit = {
+    implicit val p = context.p
+    val tlBusToGetPrefixFrom = context.locateTLBusWrapper(crossingParams.mmioBaseAddressPrefixWhere)
+    domain.element.tileHartIdNodes.foreach { case (hartid, node) =>
+      node := context.tileHartIdNodes(hartid)
+    }
+    domain.element.tileResetVectorNodes.foreach { case (hartid, node) =>
+      node := context.tileResetVectorNodes(hartid)
+    }
+  }
+
+  def connectTrace(domain: ClusterPRCIDomain, context: ClusterContextType): Unit = {
+    implicit val p = context.p
+    domain.element.traceNodes.foreach { case (hartid, node) =>
+      val traceNexusNode = BundleBridgeBlockDuringReset[TraceBundle](
+        resetCrossingType = crossingParams.resetCrossingType)
+      context.traceNodes(hartid) := traceNexusNode := node
+    }
+    domain.element.traceCoreNodes.foreach { case (hartid, node) =>
+      val traceCoreNexusNode = BundleBridgeBlockDuringReset[TraceCoreInterface](
+        resetCrossingType = crossingParams.resetCrossingType)
+      context.traceCoreNodes(hartid) :*= traceCoreNexusNode := node
+    }
+  }
+}
+
+case class ClusterAttachParams(
+  clusterParams: ClusterParams,
+  crossingParams: ElementCrossingParamsLike
+) extends CanAttachCluster
+
+case class CloneClusterAttachParams(
```

Comment:
Basically I don't like the current Clone implementation in Diplomacy, which is a little brittle, I think in the future, we can try to remove them and use `https://github.com/llvm/circt/blob/8129d5de502a8cd5d861438e99f823babc5cebb0/docs/Dialects/FIRRTL/FIRRTLAnnotations.md?plain=1#L613` to force circt do it.

Reply:
I agree the implementation of CloneLazyModule is not good right now, but I like the API of CloneLazyModule. I'm not sure what you are envisioning, but I thought the backend implementation of CloneLazyModule could change, while these APIs remain